### PR TITLE
feat(sever): for test runs, split the failed state into more descriptive statuses

### PR DESCRIPTION
**Pull request by schoren**
This PR makes the TestRun state more accurate by providing a new set of statuses: `TRACE_FAILED`, `TRIGGER_FAILED` and `ASSERTION_FAILED`.

## Changes

- add new states and implement where applicable

## Fixes

- #2285

## Checklist

- [X] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by mathnogueira — APPROVED** [review]
Looks good!

**Review by jorgeepc — COMMENTED** [review]
Looks good. One question: with this approach, is it safe to assume that if I get a `TRACE_FAILED` state, the trigger stage was correct?

**Comment by mathnogueira** [comment]
> Looks good. One question: with this approach, is it safe to assume that if I get a `TRACE_FAILED` state, the trigger stage was correct?

yes. If the trigger fails, the test execution stops

**Comment by schoren** [comment]
@jorgeepc exactly. The failed stage means that all the previous steps succedded

**Review by jorgeepc — COMMENTED** [review]
I think we need to update the OpenAPI specs.

**Comment by schoren** [comment]
@jorgeepc good catch. updated

File: server/model/tests.go [hunk 1]
```diff
@@ -180,13 +180,20 @@ const (
 	RunStateCreated             RunState = "CREATED"
 	RunStateExecuting           RunState = "EXECUTING"
 	RunStateAwaitingTrace       RunState = "AWAITING_TRACE"
-	RunStateFailed              RunState = "FAILED"
```
Comment (danielbdias) [inline server/model/tests.go:183]: Today we save this value on our database, right? How will we keep the retro compatibility with older Tracetest versions?
Should we map the old `FAILED` string as always `ASSERTION_FAILED`?

Comment (danielbdias) [reply server/model/tests.go:183]: Or even writing a migration to fix it on the database?

**Review by danielbdias — COMMENTED** [review]
It sounds ok! Before merging it I have one comment about retro compatibility: should we write a migration or handle in separate the old status to avoid problems with early versions of Tracetest?

**Pull request merged**